src/help/guides/extension-development/model-reference.md

```diff
@@ -0,0 +1,269 @@
+| `mangaId` | String | The given identifier of the Manga that owns this chapter. This should match the id of the manga that the chapter is for. |
+| `chapNum` | Number | The number of the chapter, which is used for sorting the chapter list. The number may be a decimal. Multiple chapters with the same number will be sorted based on their volume number or the group that created the chapter, if it exists. Otherwise, the chapters will be sorted on the chapter ID. |
+| `langCode` | [LanguageCode](#languagecode) | The language code which this chapter is associated with. This is used along with the `Content settings` option to filter the chapter list so only chapters for the languages specified by the user are shown. |
+
+### Optional Fields
+
+| Name | Type | Description |
+|------|------|-------------|
+| `name` | String | The title of the chapter. |
+| `volume` | Number | The volume that the chapter belongs to. It is recommended to leave the volume number out if every chapter does not have a corresponding volume number, as the volume number interferes with sorting. |
+| `group` | String | The group that posted the chapter. |
+| `time` | [Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) object | The date that the chapter is published. If omitted, the chapter will have a creation of the time when the chapter list was loaded. |
+
+## ChapterDetails
+
+The ChapterDetails object contains a small amount of metadata about the chapter and contains the list of pages.
+
+### Required Fields
+
+| Name | Type | Description |
+|------|------|-------------|
+| `id` | String | The chapter ID of the chapter that the object respresents. |
+| `mangaId` | String | The given identifier of the Manga that owns this chapter. This should match the id of the manga that the chapter is for. |
+| `pages` | String Array | The list of pages in the chapter. The pages are ordered by the index of the array, where the 0th index is the first page, and so on. Each item should be a link to the image file that represents the page. |
+| `longStrip` | Boolean | Indicates whether or not the chapter should be rendered in **Long Strip
+Mode**. A long strip is a webtoon or any other form of manga that uses very tall but narrow images. |
+
+## IconText
+
+An icon text represents a line of text and an optional icon.
+
+### Required Fields
+
+| Name | Type | Description |
+|------|------|-------------|
+| `text` | String | The text that the object represents. |
+
+### Optional Fields
+
+| Name | Type | Description |
+|------|------|-------------|
+| `icon` | String | THe icon that the object represents. |
+
+## LanguageCode
+
+A string enum defining full language names, and the corresponding language code.
+
+::: warning Warning Some full language names have typos. These may be fixed in a future version.
+:::
+
+### Values
```

Comment:
Do we really want to include these? Letting them know that the language code exists is nice. Is giving all of the possible values a waste of space? Any good IDE will list out the enum as soon as they try to use it.

If you think this is fine for readability, keeping it I guess is fine? Just something to consider.

Reply:
The model reference is supposed to be a model reference, meaning that all model values are listed. Also god forbid someone uses notepad++ or vim to fix a typo. It would be nice if this was collapsible though.

Reply:
For a collapsible component, you can use
```markdown
:::: el-collapse
::: el-collapse-item title="Lorem ipsum"
**Lorem ipsum** dolor sit amet, _consectetur_ adipiscing **elit**.
:::
::: el-collapse-item title="Pellentesque"
**Pellentesque** _habitant_ morbi tristique **senectus** et netus.
:::
::::
```

Reply:
Given how frequently instances of a word is also a navigation link (Ex: ChapterDetails usually sends you to the ChapterDetails page) - I suppose this really isn't all too much of a problem.
To keep consistency, there's really no need to collapse this. It's not like the only way a user can navigate this page is by scrolling and looking for things manually.

I don't think I mind how this is currently, and am alright with dismissing this comment.

Reply:
That seems fair.